pkg/gameservers/controller.go

```diff
@@ -939,8 +944,14 @@ func (c *Controller) syncGameServerRequestReadyState(ctx context.Context, gs *ag
 	// if the address hasn't been populated, and the Ready request comes
 	// before the controller has had a chance to do it, then
 	// do it here instead
+	var hasPodIPAddress bool
+	for _, addr := range gs.Status.Addresses {
+		if addr.Type == agonesv1.NodePodIP {
+			hasPodIPAddress = true
+		}
+	}
 	addressPopulated := false
-	if gs.Status.NodeName == "" {
+	if gs.Status.NodeName == "" || (!hasPodIPAddress && len(pod.Status.PodIPs) > 0) {
```

Comment:
So this approach means that if a `PodIP` record happens between `Starting` and `RequestReady` it won't show up until someone calls `SDK.Ready()`-- which doesn't seem ideal.

While this is good here, I am also wondering at the top of `syncGameServerStartingState` there should be an extra check of - if you are in `Starting` but your `PodIP` has yet to be populated, then let's do it now, rather than wait for RequestReady.

It may mean an extra Update on the `GameServer`, but at least it will be updated as soon as its available.

There does beget a question of "what if the `PodIP` data does not exist until after someone calls `SDK.Ready()` -- but maybe that's an edge case we cover if we have to? WDYT?

Reply:
The `syncGameServerStartingState` is maybe misleading, I've added a comment on your feedback: https://github.com/googleforgames/agones/pull/4095/files#r1987877618 

It should populate the `PodIP` during the `Starting` phase, from what I understood, I might need to deep more about the code around the [GS states](https://agones.dev/site/docs/reference/gameserver/#gameserver-state-diagram) 😄 

My guess would be that if it's still in `starting`, the `requestReady` can't be called, at least until it's in `scheduled` right ? 🤔

Reply:
> My guess would be that if it's still in `starting`, the `requestReady` can't be called, at least until it's in `scheduled` right ? 🤔

Yeah it can 😄 because it is actually scheduled. Also SDK calls are async, so you can't actually stop it. You could make it wait for a while I suppose? but then, if you don't need PodIPs, why make it wait?

Reply:
Totally make sense, thanks for the explanations 😄